### PR TITLE
Added possibility to have multiple finders

### PR DESCRIPTION
**Pull request by judovana**
Hi!

This is direct adaptation and rebase to tip of https://github.com/jenkinsci/text-finder-plugin/pull/23 , whic can be closed (unless merged first of course)

I had merged individual commits to single one - it still have sense, and to rebase it one by one was extremely painful.

### CODE REVIEW

File: src/main/java/hudson/plugins/textfinder/TextFinderPublisher.java [hunk 1]
```diff
@@ -244,6 +243,46 @@ private static Pattern compilePattern(PrintStream logger, String regexp) {
         return pattern;
     }
 
+    /** cut off first item as it is provided with getters */
```
Comment (sparkoo) [inline src/main/java/hudson/plugins/textfinder/TextFinderPublisher.java]: this comment is no longer valid. I was using only list with 0th item as primary text finder. However, now there is `primaryTextFinder` and list for others.

File: src/main/java/hudson/plugins/textfinder/TextFinderPublisher.java [hunk 2]
```diff
@@ -244,6 +243,46 @@ private static Pattern compilePattern(PrintStream logger, String regexp) {
...
+    @SuppressWarnings("unused")
+    public List<TextFinderModel> getAdditionalTextFinders() {
+        return additionalTextFinders;
+    }
+
+    // backward compatibility getters below
+    // all these getters are there for backward compatibility
+    // get first field from list which is used to store original config values
```
Comment (sparkoo) [inline src/main/java/hudson/plugins/textfinder/TextFinderPublisher.java]: same here, no longer valid comment. we're not getting 0th element, but values from `primaryTextFinder`

Comment (judovana) [reply src/main/java/hudson/plugins/textfinder/TextFinderPublisher.java]: comments removed. Wil now jump to the https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502392524

**Comment by judovana** [comment]
@basil hi! Major issues were fixed. I think you can start with review.
I'm now testing DSL api.
I would like to fix https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502392524 as separate changeset. ty in advance.

**Comment by basil** [comment]
> @basil hi! Major issues were fixed. I think you can start with review.
> I'm now testing DSL api.
> I would like to fix [#23 (comment)](https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502392524) as separate changeset. ty in advance.

A promising start. Looks like my first concern from [#23 (comment)](https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502390479) was addressed.

Now we need to address the second and third concerns from [#23 (comment)](https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502390479) and the concern from [#23 (comment)](https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502392524).

**Comment by judovana** [comment]
hi! Thank you for  eyballing it.
I would say that  also third concern is fixed, inst it?
As for  https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502392524 ,  please dont torture me. I'm going  to fix it, but it can be separate PR.

From my point of view only DSL verification is missing. I'm working on that, but am currently stuck on it. It can be seen in dsl dynamic api view, but methods the dynamic api viewer is showing do not  have sense. Was Textfinder plgin actually ever working with dsl, after the change from databound constructors to databoundsetters?  Was it wolring on all proeject types?

**Comment by judovana** [comment]
ping please?

**Comment by judovana** [comment]
@basil  Hi!

Do you mind to confirm that also third concern is fixed please?
Do you really insists on fixing the https://github.com/jenkinsci/text-finder-plugin/pull/23#issuecomment-502392524 in initial changeset? I would really like to dereffer it it to another changeset.

Can you please help with the dsl? I spent few days on it, ant it was not working before this changeset too, or I'm doing something really wrong.

Please do not let this change to rot, You could see, that there are at least four teams world wide using this feature, and so now pathcing the builds of yours on thirs own.  Thanx!

**Comment by judovana** [comment]
Dear maintainers,  collaboration with this project is really painful
Do you mind if we publish our fork on jenkinsci as new plugin?

**Comment by judovana** [comment]
https://issues.jenkins-ci.org/browse/HOSTING-896

**Comment by judovana** [comment]
Hi Basil!

You have forced us to rework the  fully backward comaptible PR to this one. And once we did it, and once we fixed even all other nits,  you stopped to communicate.  Do you have some final resolution?

Maybe we should for as jenkins/multitext-finder-plugin?

**Comment by judovana** [comment]
We have one more improvement in our for. A field, whicch based on regex can change displayName of run.  Not sure if you are interesting in any such change. If not, thane probably we really should stay forked.

**Comment by judovana** [comment]
Ok. We will continue to live in our fork, https://github.com/judovana/text-finders-plugin/ and will release it as separate plugind

**Pull request closed without merging**